### ISSUE TITLE
REST handlers silently ignore request body if unexpected

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#76694]
In https://discuss.elastic.co/t/ilm-retry-api-not-honoring-timeouts/281772 a user reported confusion over the way that Elasticsearch ignored the parameters in this request:

```
POST */_ilm/retry
{
  "timeout": "30m",
  "master_timeout": "30m"
}
```

The immediate solution was to move the parameters into the URL where they belong, but the question highlights a broader problem: if an API endpoint expects a request body then we parse the body fairly strictly and reject requests that contain unexpected things, but if the endpoint doesn't expect a body at all then we seem to leniently accept requests with a body anyway. Silently ignoring input like this is confusing, and I think we should have rejected this request with a `400 Bad request` instead.

**Comment by rjernst** [comment]
What version was this on? That should have been fixed already in 7.0: https://github.com/elastic/elasticsearch/pull/37504

**Comment by DaveCTurner** [comment]
Bah sorry for the noise, I guess they must have been using 6.8.x. It does look to be fixed indeed.
